16/umbraco-cms/customizing/extending-overview/extension-types/kind.md

```diff
@@ -80,32 +76,46 @@ In this example, the Header App extension uses the `kind: 'button'`, meaning it
 
 Here’s an example of how to register and use the Button Kind in a Header App extension:
 
+{% code title="kinds/manifests.ts" %}
+
+{% hint style="warning" %}
+This example uses the dynamic extension registration approach. `umbExtensionsRegistry.register()` might be called from within an entrypoint lifecycle method like `onInit()`. For more information, see the [Backoffice Entry Point](./backoffice-entry-point.md) article.
+{% endhint %}
+
 ```typescript
 import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import type { ManifestKind } from "@umbraco-cms/backoffice/extension-api";
 
-const manifest: UmbExtensionManifest = {
+const manifest: ManifestKind<UmbExtensionManifest> = {
   type: 'kind',
   alias: 'Umb.Kind.MyButtonKind',  // Alias for the Kind
   matchType: 'headerApp', // Extension type the Kind applies to
-  matchKind: 'button',  // Defines the Kind alias
+  matchKind: 'customHeaderAppButton',  // Defines the Kind alias
   manifest: {
     elementName: 'umb-header-app-button',
   },
 };
 
 umbExtensionsRegistry.register(manifest);
 ```
+{% endcode %}
 
-This code registers the Button Kind, so other Header App extensions using `type: 'headerApp'` and `kind: 'button'` will inherit the preset `elementName: 'umb-header-app-button'`.
+This code registers the Button Kind, so other Header App extensions using `type: 'headerApp'` and `kind: 'customHeaderAppButton'` will inherit the preset `elementName: 'umb-header-app-button'`.
 
-Now, another Header App extension can be created without defining `elementName`, as it will automatically inherit it from the Kind:
+Now another Header App extension can be created without defining `elementName`, as it will automatically inherit it from the Kind:
+
+{% code title="kinds/manifests.ts" %}
+
+{% hint style="warning" %}
+This example uses the dynamic extension registration approach. `umbExtensionsRegistry.register()` might be called from within an entrypoint lifecycle method like `onInit()`. For more information, see the [Backoffice Entry Point](./backoffice-entry-point.md) article.
+{% endhint %}
```

Comment:
Maybe we don't need this twice since you mention it above, so think you can remove.